### PR TITLE
Close #321.

### LINKED ISSUE #321: Replace OR-Tools
Given how many build issues we've encountered with OR-Tools, we should look at some alternative linear solver libraries. The only code changes involved would be to re-implement `hermes::MinimizeIoTimePlacement`.

### PR DESCRIPTION
**Pull request by hyoklee**
Replace OR-TOOLS with GLPK.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -134,7 +134,7 @@ LSAN_OPTIONS=suppressions=${CMAKE_CURRENT_SOURCE_DIR}/data/asan.supp)
 #------------------------------------------------------------------------------
 add_executable(bo buffer_organizer_test.cc)
 target_link_libraries(bo ${LIBRT} hermes MPI::MPI_CXX
-  $<$<BOOL:${HERMES_RPC_THALLIUM}>:thallium>)
+  $<$<BOOL:${HERMES_RPC_THALLIUM}>:thallium> glog::glog gflags)
```
Comment (ChristopherHogan) [inline test/CMakeLists.txt:137]: Why does this test depend on gflags?

File: CMakeLists.txt [hunk 2]
```diff
@@ -270,6 +270,22 @@ if(HERMES_RPC_THALLIUM)
...
+# GLOG
+find_package(glog REQUIRED)
+if(glog_FOUND)
+  message(STATUS "found GLOG at ${glog_DIR}")
+endif()
+find_path(GLOG_INCLUDE_DIRS NAME glog PATH_SUFFIXES include/)
+include_directories(${GLOG_INCLUDE_DIRS})
+
+# GFLAGS
+find_package(gflags REQUIRED)
```
Comment (ChristopherHogan) [inline CMakeLists.txt:282]: I would prefer to make gflags an optional dependency, or remove it altogether. I don't use it for anything, and glog works fine without it.

File: src/data_placement_engine.cc [hunk 3]
```diff
@@ -272,103 +266,141 @@ Status MinimizeIoTimePlacement(const std::vector<size_t> &blob_sizes,
...
+    glp_set_row_bnds(lp, i+1, GLP_FX, 1.0, 1.0);
 
     // TODO(KIMMY): consider remote nodes?
     for (size_t j {0}; j < num_targets; ++j) {
+      int ij = i * num_targets + j + 1;
       std::string var_name {"blob_dst_" + std::to_string(i) + "_" +
                             std::to_string(j)};
-      blob_fraction[i][j] = solver.MakeNumVar(0.0, 1, var_name);
-      blob_constrt[num_constrts+i]->SetCoefficient(blob_fraction[i][j], 1);
+      // total umber of variables = number of blobs * number of targets.
```
Comment (ChristopherHogan) [inline src/data_placement_engine.cc:294]: 'umber' -> 'number'

**Review by ChristopherHogan — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (hyoklee) [reply CMakeLists.txt:282]: I think glog depends on gflags.
```
/home/runner/work/hermes/hermes/spack/opt/spack/linux-ubuntu20.04-x86_64_v4/gcc-10.3.0/glog-0.4.0-jcohf3eh7cax7d6742d6owabbwdj6464/include/glog/logging.h:86:10: fatal error: gflags/gflags.h: No such file or directory
   86 | #include <gflags/gflags.h>
      |          ^~~~~~~~~~~~~~~~~
```

Comment (ChristopherHogan) [reply CMakeLists.txt:282]: You can disable the gflags dependency [via cmake](https://github.com/google/glog/blob/master/CMakeLists.txt#L41). Unfortunately it looks like the [glog spack package](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/glog/package.py) doesn't expose a variant without gflags. It hard-codes the dependency. I guess we're stuck with it for now.

**Review by ChristopherHogan — APPROVED** [review]

**Pull request merged**